app/scripts/controllers/mobileClients.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:MobileClientsController
+ * @description
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('MobileClientsController',
+    function ($scope,
+              $filter,
+              $routeParams,
+              Constants,
+              DataService,
+              ProjectsService) {
+
+      var ctrl = this;
+      ctrl.projectName = $routeParams.project;
+      ctrl.emptyMessage = "Loading...";
+      ctrl.alerts = {};
+      ctrl.imagesByDockerReference = {};
+      ctrl.redirectUrl = "/project/" + ctrl.projectName + "/overview";
+      ctrl.breadcrumbs = [
+        {
+          title: "Mobile Clients",
+          link: "project/" + ctrl.projectName + "/browse/mobile-clients"
+        },
+        {
+          title: $routeParams.mobileclient
+        }
+      ];
+
+      var watches = [];
+
+      ProjectsService
+        .get($routeParams.project)
+        .then(_.spread(function(project, context) {
+          ctrl.project = project;
+          ctrl.projectContext = context;
+
+          DataService.get(Constants.MOBILE_CLIENT_VERSION, $routeParams.mobileclient, context, { errorNotification: false }).then(
+            // success
+            function(mobileClient) {
+              ctrl.loaded = true;
+              ctrl.mobileClient = mobileClient;
+
+              // If we found the item successfully, watch for changes on it
+              watches.push(DataService.watchObject(Constants.MOBILE_CLIENT_VERSION, $routeParams.mobileclient, context, function(mobileClient, action) {
```

Comment:
Wondering if there is any way of reusing existing watches? Seems wasteful to always be creating new ones? Not something that needs to change for this PR but just wondering if there are any examples of watches being reused in the code base

Reply:
good point, I ll test this on minishift

Reply:
I believe `Dataservice` manages the websockets. Only applying one per key

Reply:
I've used this controller as an example https://github.com/openshift/origin-web-console/blob/master/app/scripts/controllers/deployment.js#L83
this watcher is being destroyed as soon as user leaves this view